metadata-ingestion/src/datahub/ingestion/source/dynamodb/dynamodb.py

```diff
@@ -407,13 +433,20 @@ def construct_schema_metadata(
         canonical_schema: List[SchemaField] = []
         schema_size = len(schema.values())
         table_fields = list(schema.values())
-
         if schema_size > MAX_SCHEMA_SIZE:
             # downsample the schema, using frequency as the sort key
             self.report.report_warning(
                 key=dataset_urn,
                 reason=f"Downsampling the table schema because MAX_SCHEMA_SIZE threshold is {MAX_SCHEMA_SIZE}",
             )
+            # schema will be sorted by count and delimited_name in descending order and sliced to only include MAX_SCHEMA_SIZE items
+            table_fields = sorted(
+                schema.values(),
+                key=lambda x: (
+                    -x["count"],
+                    x["delimited_name"],
```

Comment:
it looks like we're already sorting a few lines down - can we just do the more sophisticated sort below instead?

Reply:
Thank you for your suggestion! That sounds good to me, it looks like same sophisticated sort can apply to MongoDB as well, as this is referred from `mongodb.py`, I can update this same change in my MongoDB PR as well